services/ymax-planner/src/engine.ts

```diff
@@ -220,6 +220,8 @@ const processPortfolioEvents = async (
       { cosmosRest, spectrum },
     );
     const plannerContext = {
+      // @ts-expect-error "amount" is not present on all varieties of FlowDetail
+      amount: flowDetail.amount,
```

Comment:
any particular reason to `@ts-expect-error` rather than a typesafe expression?

```suggestion
      amount: 'amount' in flowDetail ? flowDetail.amount : undefined,
```

Reply:
That doesn't help; `planDepositToAllocations` and `planWithdrawFromAllocations` require `amount` to be a NatAmount but the above construction (just like the shorter one I'm using) allows for it to be undefined.